pkg/codeconfig/function.go

```diff
@@ -79,6 +83,25 @@ func (a *Api) AddWorker(worker *pb.ApiWorker) error {
 	return nil
 }
 
+func (a *Api) AddSecurityDefinition(name string, sd *pb.ApiSecurityDefinition) {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	a.securityDefinitions[name] = sd
+}
+
+func (a *Api) AddSecurity(name string, scopes []string) {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	sc := []string{}
+	if scopes != nil {
+		sc = scopes
+	}
+
+	a.security[name] = sc
```

Comment:
```suggestion
	if scopes != nil {
  	     a.security[name] = sc
	}
```

Reply:
should `sc` be `scopes` in this change as well? Otherwise looks good :+1:

Reply:
yip, sorry. i updated the suggestion..

Reply:
just looking at this again the main difference is that `a.security[name]` will always default to `[]string{}` instead of being nil (scopes cannot be nil according the the spec and this prevent the security rule being removed due to `omitEmpty`)

Reply:
Have cleaned this up, with an empty slice default.